README.md

```diff
@@ -25,41 +29,64 @@ Check out our get started with `tapyr` [blog post](www.appsilon.com/post/introdu
 ### Using Devcontainer
 
 To ensure a consistent development experience across all environments, we recommend using the [devcontainer](https://code.visualstudio.com/docs/remote/containers) configuration with Visual Studio Code or DevPod for container-based development.
+If you prefer a local setup, follow the instructions below.
 
 1. **Start the Devcontainer**: Open the project in VS Code and select "Reopen in Container" when prompted, or use the Command Palette (`Ctrl+Shift+P`) and choose "Remote-Containers: Reopen in Container". Alternatively, use [DevPod](https://devpod.sh/) following their instructions.
 2. **Activate the virtual environment**:
    ```sh
-   poetry shell
+   source .venv/bin/activate
    ```
 3. **Run the application**:
    ```sh
    shiny run app.py --reload
    ```
 4. **Execute tests**:
    ```sh
-   poetry run pytest
+   pytest
    ```
 
 *Note*: The Devcontainer might limit some `playwright` features, such as `codegen`. For full functionality, consider a local setup.
 
-### Setting Up Locally with Poetry
+### Setting Up Locally with `uv`
 
 For developers preferring a local setup without Devcontainer:
 
-1. **Install pipx**: Ensure pipx is installed for managing isolated CLI apps.
-2. **Install Poetry**:
+1. **Install `uv`**: [Follow the installation guide](https://docs.astral.sh/uv/getting-started/installation/).
+2. **Clone the repository** and navigate to it.
+3. **Install dependencies**:
    ```sh
-   pipx install poetry
+   uv sync
+   uv run playwright install --with-deps
    ```
-3. **Clone the repository** and navigate to it.
-4. **Install dependencies**:
+4. **Run the application**:
    ```sh
-   poetry install
-   playwright install
+   uv run shiny run app.py --reload
    ```
 
 *Attention*: Follow any additional steps prompted by `playwright install`.
 
+More information on `uv` can be found in the [official documentation](https://docs.astral.sh/uv/).
+
+### Setting Up Locally with `pip`
+
+Although not recommended, you can set up the project using `pip`:
+
+1. **Clone the repository** and navigate to it.
+2. **Create a virtual environment**:
+   ```sh
+   python -m venv .venv
+   source .venv/bin/activate
+   ```
+3. **Install package with dependencies**:
+   ```sh
+   pip install -e ."[dev]"
+   playwright install --with-deps
+   ```
+4. **Run the application**:
+   ```sh
+   shiny run app.py --reload
+   ```
+
 ### Deployment on Posit Connect
 
 Deploy your application to Posit Connect by:
```

Comment:
I would also add a small note here that the user needs to first have run `source .venv/bin/activate`.

It might seem like a no brainer, but from a fresh user perspective they might wonder why nothing is working? (Yes, I've had first hand encounter with this situation 😄 )